src/components/PopupWithForms.js

```diff
@@ -1,14 +1,23 @@
 import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
-  constructor(popupSelector, handleFormSubmit, formSelector) {
+  constructor(popupSelector, handleFormSubmit, settings) {
     super({ popupSelector });
     this._handleFormSubmit = handleFormSubmit;
-    this._popupForm = this._popupElement.querySelector(formSelector);
+    this._button = this._popupElement.querySelector(
+      settings.submitButtonSelector
+    );
+    this._buttonText = this._button.textContent;
+    this._popupForm = this._popupElement.querySelector(settings.formSelector);
     this._inputList = Array.from(
       this._popupForm.querySelectorAll(".modal__input")
     );
-    this._submitButton = this._popupForm.querySelector(".modal__button");
+  }
+
+  renderSaving(isSaving) {
+    isSaving
+      ? (this._button.textContent = "Saving...")
```

Comment:
**Could be improved**


`"Saving..."` - is a hardcode. You can change it like that


```js

  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```